### PR TITLE
Qa

### PR DESCRIPTION
**Pull request by gdhungana**
This PR extends the flavor types for QAs as defined in newexp_desi. Also adds a mean sky continuum QA.

### CODE REVIEW

**Comment by sbailey** [comment]
The travis tests are failing because they need a more recent version of desiutil than is specified in the .travis.yml.  @profxj or @gdhungana please update the .travis.yml to use desiutil 1.3.4 to get the desiutil.stats module.

@profxj should be the one to approve this PR since it is going into his branch.

**Comment by gdhungana** [comment]
I was expecting that and am letting @profxj to check that. On my side, I did not merge the master but the desispec.qa_fluxing branch before sending the PR, to make my branch consistent with that. So as this gets reviewed and merged, hopefully it can be updated once before finally merging to the master.

File: py/desispec/sky.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def qa_skysub(param, frame, fibermap, skymodel):
     specmin, specmax = np.min(frame.fibers), np.max(frame.fibers)
     skyfibers=np.where((fibermap["OBJTYPE"]=="SKY")&
         (fibermap["FIBER"]>=specmin)&(fibermap["FIBER"]<=specmax))[0]
-    assert np.max(skyfibers) < 500
+    assert np.max(skyfibers) < 500 # only spectrograph 0??
```
Comment (profxj) [inline py/desispec/sky.py]: This is only insuring that there are fewer than 500 fibers.
These are indices, not the fiber values

**Comment by profxj** [comment]
I think the changes all look good.  
There are going to be some issues with what I've been building, 
but I'll deal with them.

I'm going to merge.

**Comment by sbailey** [comment]
You merged before fixing the travis test failure.  Please never merge into master if the Travis tests are failing, even if they pass on your laptop.  They could be hiding some other environment-dependent bug, or a trivial failure like wrong desiutil version could mask a more serious real error.

I just pushed the .travis.yml directly to master to fix the these there (I hope).

**Comment by profxj** [comment]
Crud.  I didn’t mean to merge into master.
I thought I was only merging into the qa_fluxing branch.

Sorry,

X



> On Mar 2, 2016, at 3:06 PM, sbailey <notifications@github.com> wrote:
> 
> You merged before fixing the travis test failure. Please never merge into master if the Travis tests are failing, even if they pass on your laptop. They could be hiding some other environment-dependent bug, or a trivial failure like wrong desiutil version could mask a more serious real error.
> 
> I just pushed the .travis.yml directly to master to fix the these there (I hope).
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/desihub/desispec/pull/129#issuecomment-191484761>.
> 
> !DSPAM:826,56d7723b28493143511550!

----------------------------------------------
J. Xavier Prochaska
UCO/Lick Observatory
1156 High St.
UC Santa Cruz
Santa Cruz, CA 95064

http://www.ucolick.org/~xavier/
http://imps.ucolick.org/
http://www.qpqsurvey.org/
http://k1dm3.ucolick.org/
http://www.ucolick.org/~xavier/IDL/index.html
https://github.com/pyigm/pyigm
https://github.com/linetools/linetools

831-459-2135 (Direct)
831-459-2991 (UCO/Lick Main)
831-459-5244 (Fax)

**Comment by sbailey** [comment]
For the record, you did do the right thing here of merging into the qa_fluxing branch and not master.  Now I'm mystified how the tests got broken on master without us noticing.

**Comment by sbailey** [comment]
I think this was user error on my part.  The Travis build log claims master has always been working:

https://travis-ci.org/desihub/desispec/builds

After your merge, I went to the desispec main page:

https://github.com/desihub/desispec

and saw the "Build Failure" tag and assumed it was refering to master, not some branch.  I followed the link and it took me to the failing build, but I didn't notice that it wasn't master.

It appears that the travis build status image is referring to whatever the latest build was, not whatever the build is for the branch that we are looking at on github.  Beware.

**Comment by profxj** [comment]
Phew!

This makes much more sense..

----------------------------------------------
J. Xavier Prochaska
UCO/Lick Observatory
1156 High St.
UC Santa Cruz
Santa Cruz, CA 95064

http://www.ucolick.org/~xavier/
http://imps.ucolick.org/
http://www.qpqsurvey.org/
http://k1dm3.ucolick.org/
http://www.ucolick.org/~xavier/IDL/index.html
https://github.com/profxj/xastropy

831-459-2135 (Direct)
831-459-2991 (UCO/Lick Main)
831-459-5244 (Fax)

**Pull request merged**